### PR TITLE
Webpack 5 support

### PR DESCRIPTION
**Pull request by max-degterev**
Changelog:

- Updated dependencies to latest versions
- Removed loader-utils dependency (no longer needed)
- Moved asset creation from `compiler.hooks.emit` to `compilation.hooks.processAssets` at `PROCESS_ASSETS_STAGE_DERIVED`
- Create new asset with `RawSource` instead of by hand for better Webpack compatibility
- Fixed a case when in watch mode a syntax error inside a translation module would crash Webpack process
- Fixed dictionaries going in a rebuild loop in certain race conditions
- Fixed dictionaries being re-written with results of previous successful compilation in case of errors, now fails hard

### CODE REVIEW

File: fixtures/webpack.config.js [hunk 1]
```diff
@@ -28,11 +28,7 @@ module.exports = {
   plugins: [
     new CleanWebpackPlugin(),
     new I18nModular({ emitFile }),
-    new ManifestPlugin(),
-    new CompressionPlugin({
```
Comment (peterhass) [inline fixtures/webpack.config.js:32]: compression not needed anymore?

**Review by peterhass — APPROVED** [review]

File: fixtures/webpack.config.js (hunk 1 above)
Comment (max-degterev) [reply fixtures/webpack.config.js:32]: no and we also remove it from our builds. the point here was to test an example config, it doesn't really matter if we test compression plugin as well. but maybe I add it again just to see that jsons are not being gzipped, that would be annoying

Comment (max-degterev) [reply fixtures/webpack.config.js:32]: added it back

File: lib/plugin.js [hunk 2]
```diff
@@ -47,80 +49,95 @@ class I18nModularPlugin {
     const keysRoot = rebase(compiler.context, relativeKeysRoot);
     const dictionaryRegex = getDictionaryRegex(rebase(compiler.context, dictionaryPattern));
 
+    // Stops watching dictionary files to avoid rebuild loops in watch mode
```
Comment (peterhass) [inline lib/plugin.js:52]: maybe migrate some of those comments over to functions with appropriate names?

**Review by peterhass — APPROVED** [review]

File: lib/plugin.js (hunk 2 above)
Comment (max-degterev) [reply lib/plugin.js:52]: instance methods perhaps, but still would need too many arguments

Comment (peterhass) [reply lib/plugin.js:52]: Would rather have a function with many arguments vs comments that get outdated

Comment (max-degterev) [reply lib/plugin.js:52]: treat it as a step towards JSDoc!

Comment (peterhass) [reply lib/plugin.js:52]: JSno 😔

**Pull request merged**